### ISSUE TITLE
[API] Create DSL for initialising a model

### ISSUE THREAD

**Issue by dosier** [JetBrains/KotlinDL#133]
This would be great for initialising additional fields of the model.

That way I could do e.g. 
```kotlin
    val model = sequential {
        name = "fancy model"
        layers {
            +Input(..),
            +Conv1D(..)
        }
    }
```

Instead of 
```kotlin
    private val model = Sequential.of(
        Input(..),
        Conv2D(..)
    ).apply {
        name = "fancy model"
    }
```

**Comment by zaleslaw** [comment]
I'd like your ideas @dosier , so I suppose Kotlin could give us more for creating and describing the neural networks and creating a special DSL at the top of the "Keras API" then it will be finished and stabilized during the 0.4 release, probably

**Comment by smallshen** [comment]
https://github.com/smallshen/KotlinDL/blob/baffecf2490632c9607b6049f2dfeee33d1915a9/api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/extension/dsl/GraphTrainableModelDslTest.kt
Just a little dsl builder, it should be just what you expect

https://github.com/JetBrains/KotlinDL/pull/252

**Comment by zaleslaw** [comment]
Thanks, @smallshen for your PR #252. It's useful for sequential models, I agree.
Do you have any ideas on how to apply this kind of DSL to the Functional API (graph where nodes could have multiple inputs/outputs)?

**Comment by zaleslaw** [comment]
This issue is closed in #309
